Description: Remember the maximization state of the window.
Author: Simon Quigley <tsimonq2@ubuntu.com>
Origin: upstream
Bug: https://github.com/lxqt/qterminal/issues/450
Bug-Ubuntu: https://launchpad.net/bugs/1754496
Applied-Upstream: commit:14b1ee0
Last-Update: 2018-07-20
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -193,6 +193,8 @@ MainWindow *QTerminalApp::newWindow(bool
     else
     {
         window = new MainWindow(cfg, dropMode);
+        if (Properties::Instance()->windowMaximized)
+            window->setWindowState(Qt::WindowMaximized);
         window->show();
     }
     return window;
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -542,6 +542,7 @@ void MainWindow::closeEvent(QCloseEvent
             if (Properties::Instance()->saveSizeOnExit) {
             	Properties::Instance()->mainWindowSize = size();
             }
+            Properties::Instance()->windowMaximized = isMaximized();
             Properties::Instance()->mainWindowState = saveState();
         }
         Properties::Instance()->saveSettings();
@@ -575,6 +576,7 @@ void MainWindow::closeEvent(QCloseEvent
         Properties::Instance()->mainWindowSize = size();
         Properties::Instance()->mainWindowState = saveState();
         Properties::Instance()->askOnExit = !dontAskCheck->isChecked();
+        Properties::Instance()->windowMaximized = isMaximized();
         Properties::Instance()->saveSettings();
         for (int i = consoleTabulator->count(); i > 0; --i) {
             consoleTabulator->removeTab(i - 1);
--- a/src/properties.cpp
+++ b/src/properties.cpp
@@ -147,6 +147,8 @@ void Properties::loadSettings()
 
     confirmMultilinePaste = m_settings->value("ConfirmMultilinePaste", false).toBool();
     trimPastedTrailingNewlines = m_settings->value("TrimPastedTrailingNewlines", false).toBool();
+
+    windowMaximized = m_settings->value("LastWindowMaximized", false).toBool();
 }
 
 void Properties::saveSettings()
@@ -240,9 +242,10 @@ void Properties::saveSettings()
     m_settings->setValue("ChangeWindowIcon", changeWindowIcon);
     m_settings->setValue("enabledBidiSupport", enabledBidiSupport);
 
-
     m_settings->setValue("ConfirmMultilinePaste", confirmMultilinePaste);
     m_settings->setValue("TrimPastedTrailingNewlines", trimPastedTrailingNewlines);
+
+    m_settings->setValue("LastWindowMaximized", windowMaximized);
 }
 
 void Properties::migrate_settings()
@@ -312,6 +315,7 @@ void Properties::migrate_settings()
 	    geom.restoreGeometry(settings.value("MainWindow/geometry").toByteArray());
             settings.setValue("MainWindow/size", geom.size());
             settings.setValue("MainWindow/pos", geom.pos());
+            settings.setValue("MainWindow/isMaximized", geom.isMaximized());
             settings.remove("MainWindow/geometry");
 	}
     }
--- a/src/properties.h
+++ b/src/properties.h
@@ -106,6 +106,8 @@ class Properties
         bool confirmMultilinePaste;
         bool trimPastedTrailingNewlines;
 
+        bool windowMaximized;
+
 
     private:
 
